core/embed/rust/src/ui/component/base.rs

```diff
@@ -4,6 +4,8 @@ use heapless::Vec;
 
 #[cfg(feature = "model_t1")]
 use crate::ui::model_t1::event::ButtonEvent;
+#[cfg(feature = "model_tr")]
```

Comment:
instead: `#[cfg(any(feature = "model_t1", feature = "model_tr"))]`
dtto in all ButtonEvent cases

Reply:
Yeah, it would be the cleanest, but it is not possible, as the directory where the `event.rs` file lies is also different - `crate::ui::model_t1` vs `crate::ui::model_tr`.

During compilation, it would complain that in the case of `model_tr` feature the `model_t1` directory does not exist

Reply:
That brings the idea of putting it into some shared directory...

Reply:
right, I didn't notice that, sorry.

Yes, shared definitions are the way to go. In fact, it should be something like `ui/event.rs` defining both Button and Touch events. We might not even need these to be conditional on model, as long as the usages are conditional. What do you think @mmilata ?

Reply:
+1 for sharing

Reply:
Shared `event.rs` created in [c042b45](https://github.com/trezor/trezor-firmware/pull/2230/commits/c042b4567c0ced0072ad166c7b12323cbdd1d6d9)

I am keeping the conditional use based on the model, otherwise, it would be complaining about unused imports